### PR TITLE
The Golden Bolt Returns, but weaker

### PR DESCRIPTION
**Pull request by Epsilos**
## About The Pull Request

Bad timing but this isn't a joke PR
Adds tasers to the starting loadout of the security officer and the warden. Also adds a supply crate of 2 tasers for 4000 points.

In addition, taser electrodes have been nerfed to be somewhat like stunbatons. At 1 hit, you're confused and jittery. At 2 hits, you drop into hardstun after a short delay. At 3 hits, immediate hardstun. This allows you some time to fight back against your aggressor and also adds more meaning to the low energy count of a single taser.

## Why It's Good For The Game

The absence of the golden bolt leaves a hole in my heart, and in my opinion disablers are lower on the totem pole than the crew literally throwing a morgue corpse at someone to down them immediately. This should add a meaningful sidegrade to the disabler that stays in line with its sister weapon, the baton.

## Changelog
:cl:
add: Security officers and Wardens have a taser again! The HoS just has his special gun as usual.
add: Cargo can now order tasers again! For 4000 points for two!
balance: Tasers are no longer an instant spessman go horizontal, and should act more like stunbatons.
/:cl:

### CODE REVIEW

**Comment by triplezeta** [comment]
haha funny PR for Coder Fools! Wait what do you mean its not a joke PR

**Comment by triplezeta** [comment]
In seriousness, tasers in their current state are incredibly unhealthy for the game. They can instantly end an antag round, or conversely, enable an antag to breeze through the entire crew (even more so now with weapon powercells). 

I'd recommend either reducing the shot capacity of the taser, making tasers not do a hardstun (perhaps an effect like a stun baton, but weaker?) or both

**Comment by dinnerwoerror** [comment]
haha pognt lieutenants stun revolver

**Comment by triplezeta** [comment]
also if you could give the hos his taser mode back thatd be really funny (keep the ion too)

**Comment by dinnerwoerror** [comment]
hey ass key, remember those multiple dozens of changeling rounds on beestation where the ling would get their hands on the head of security's taser gun and the fire axe and would be going from department to department, using the combination of the two weapons to zap and decapitate any crew members they saw, autism screeching any seccie's energy guns and fleshmending any cargonian bullets away in the process

**Comment by triplezeta** [comment]
once i killed a ling and dragged them to the cremator but they did the funny revival before i could hit the button, but the first thing they tried to do was pull out the hos gun on me while they were up against a solid object so i shoved them down and hit them with the golden bolt

**Comment by Superyodeler** [comment]
Make lings spawn with a taser

**Comment by dinnerwoerror** [comment]
nah don't they get the overpowered as shit autism screech

**Comment by Epsilos** [comment]
this'll take a fuckton of testing (or maybe just one session) but i'm going to make taser electrodes do the same nerfed, built-up stun the stunbaton does

**Comment by Epsilos** [comment]
it now takes 3 shots to down immediately. I don't know if 2 is enough to hardstun because I can't test that solo with much accuracy before I go to bed today. It can ship, though.

**Comment by dinnerwoerror** [comment]
so it's like citadel code tasers

**Comment by triplezeta** [comment]
3 shots to down immediately is one shot less than with a disabler. Considering tasers have only 5 shots, pretty balanced.

**Comment by Epsilos** [comment]
preparing to replace tasers with hybrids in code since someone brought that up and it'd save an inv space for the affected jobs

**Comment by Epsilos** [comment]
since i couldn't for the life of me make the hoslaser actually display its ammo in stun mode, the hoslaser is not getting a stun mode for the time being, pending someone respriting it and magically making that work.

File: code/modules/cargo/packs.dm [hunk 1]
```diff
@@ -277,6 +277,15 @@
...
 
+/datum/supply_pack/security/taser
+	name = "Hybrid Taser Crate"
+	desc = "Two disabler-taser hybrid weapons. Requires Security access to open."
+	cost = 4000
+	contains = list(/obj/item/gun/energy/e_gun/advtaser,
+					/obj/item/gun/energy/e_gun/advtaser)
+	crate_name = "hybrid taser crate"
+	dangerous = TRUE
+
```
Comment (Superyodeler) [inline code/modules/cargo/packs.dm:288]: This should be put in the cargo file under the waspstation folder.

File: code/modules/jobs/job_types/security_officer.dm [hunk 2]
```diff
@@ -126,7 +126,7 @@ GLOBAL_LIST_INIT(available_depts, list(SEC_DEPT_ENGINEERING, SEC_DEPT_MEDICAL, S
 	shoes = /obj/item/clothing/shoes/jackboots
 	l_pocket = /obj/item/restraints/handcuffs
 	r_pocket = /obj/item/assembly/flash/handheld
-	suit_store = /obj/item/gun/energy/disabler
+	suit_store = /obj/item/gun/energy/e_gun/advtaser
```
Comment (Superyodeler) [inline code/modules/jobs/job_types/security_officer.dm:129]:
```suggestion
	suit_store = /obj/item/gun/energy/e_gun/advtaser		// Wasp edit - Readds tasers
```

File: code/modules/jobs/job_types/warden.dm [hunk 3]
```diff
@@ -47,7 +47,7 @@
 	glasses = /obj/item/clothing/glasses/hud/security/sunglasses
 	r_pocket = /obj/item/assembly/flash/handheld
 	l_pocket = /obj/item/restraints/handcuffs
-	suit_store = /obj/item/gun/energy/disabler
+	suit_store = /obj/item/gun/energy/e_gun/advtaser
```
Comment (Superyodeler) [inline code/modules/jobs/job_types/warden.dm:50]:
```suggestion
	suit_store = /obj/item/gun/energy/e_gun/advtaser		// Wasp edit - Readds tasers
```

File: code/modules/projectiles/projectile/energy/stun.dm [hunk 4]
```diff
@@ -3,27 +3,53 @@
...
-	paralyze = 100
-	stutter = 5
-	jitter = 20
 	hitsound = 'sound/weapons/taserhit.ogg'
 	range = 7
 	tracer_type = /obj/effect/projectile/tracer/stun
 	muzzle_type = /obj/effect/projectile/muzzle/stun
 	impact_type = /obj/effect/projectile/impact/stun
 
+	var/confusion_amt = 10
```
Comment (Superyodeler) [inline code/modules/projectiles/projectile/energy/stun.dm:12]:
```suggestion
	var/confusion_amt = 10		// Wasp Edit Begin - Nerfs tasers
```

File: code/modules/projectiles/projectile/energy/stun.dm [hunk 5]
```diff
@@ -3,27 +3,53 @@
...
+	var/stamina_loss_amt = 60
+	var/apply_stun_delay = 2 SECONDS
+	var/stun_time = 5 SECONDS
+
+	var/attack_cooldown_check = 0 SECONDS
+	var/attack_cooldown = 2.5 SECONDS
```
Comment (Superyodeler) [inline code/modules/projectiles/projectile/energy/stun.dm:18]:
```suggestion
	var/attack_cooldown = 2.5 SECONDS		// Wasp Edit Begin - Nerfs tasers
```

File: code/modules/projectiles/projectile/energy/stun.dm [hunk 6]
```diff
@@ -3,27 +3,53 @@
...
 		do_sparks(1, TRUE, src)
 	else if(iscarbon(target))
 		var/mob/living/carbon/C = target
-		SEND_SIGNAL(C, COMSIG_ADD_MOOD_EVENT, "tased", /datum/mood_event/tased)
-		SEND_SIGNAL(C, COMSIG_LIVING_MINOR_SHOCK)
 		if(C.dna && C.dna.check_mutation(HULK))
 			C.say(pick(";RAAAAAAAARGH!", ";HNNNNNNNNNGGGGGGH!", ";GWAAAAAAAARRRHHH!", "NNNNNNNNGGGGGGGGHH!", ";AAAAAAARRRGH!" ), forced = "hulk")
 		else if((C.status_flags & CANKNOCKDOWN) && !HAS_TRAIT(C, TRAIT_STUNIMMUNE))
-			addtimer(CALLBACK(C, /mob/living/carbon.proc/do_jitter_animation, jitter), 5)
+			C.Jitter(20)
```
Comment (Superyodeler) [inline code/modules/projectiles/projectile/energy/stun.dm:29]:
```suggestion
			C.Jitter(20)				// Wasp Edit Begin - Nerfs tasers
```

File: code/modules/projectiles/projectile/energy/stun.dm [hunk 7]
```diff
@@ -3,27 +3,53 @@
...
+		
+
+/obj/projectile/energy/electrode/proc/apply_stun_effect_end(mob/living/target)
+	var/trait_check = HAS_TRAIT(target, TRAIT_STUNRESISTANCE) //var since we check it in out to_chat as well as determine stun duration
+	if(trait_check)
+		target.Knockdown(stun_time * 0.1)
+	else
+		target.Knockdown(stun_time)
+	if(!target.IsKnockdown())
+		to_chat(target, "<span class='warning'>Your muscles seize, making you collapse[trait_check ? ", but your body quickly recovers..." : "!"]</span>")
```
Comment (Superyodeler) [inline code/modules/projectiles/projectile/energy/stun.dm:51]:
```suggestion
		to_chat(target, "<span class='warning'>Your muscles seize, making you collapse[trait_check ? ", but your body quickly recovers..." : "!"]</span>")		// Wasp Edit End
```

File: code/modules/projectiles/projectile/energy/stun.dm [hunk 8]
```diff
@@ -3,27 +3,53 @@
...
 	muzzle_type = /obj/effect/projectile/muzzle/stun
 	impact_type = /obj/effect/projectile/impact/stun
 
+	var/confusion_amt = 10		// Wasp Edit Begin - Nerfs tasers
+	var/stamina_loss_amt = 60
+	var/apply_stun_delay = 2 SECONDS
+	var/stun_time = 5 SECONDS
+
+	var/attack_cooldown_check = 0 SECONDS
+	var/attack_cooldown = 2.5 SECONDS		// Wasp Edit Begin - Nerfs tasers
```
Comment (Superyodeler) [inline code/modules/projectiles/projectile/energy/stun.dm:18]: Looks like I made a mistake here, this should be an ending comment.
```suggestion
	var/attack_cooldown = 2.5 SECONDS		// Wasp Edit End
```

**Comment by dinnerwoerror** [comment]
haha, stale!

**Comment by Maldaris** [comment]
This has been test merged for awhile and nobody has complained/we've not seen major errors or abuses. Fullmerging.

**Pull request merged**